libs/blocks/global-navigation/global-navigation.js

```diff
@@ -405,19 +443,21 @@ class Gnav {
   };
 
   decorateBrand = () => {
-    const brandBlock = this.body.querySelector('[class^="gnav-brand"]');
-    if (!brandBlock) return null;
-    const imgRegex = /(\.png|\.svg|\.jpg|\.jpeg)$/;
+    const brandBlock = this.body.querySelector('.gnav-brand');
+    if (!brandBlock) return '';
+
+    const imgRegex = /(\.png|\.svg|\.jpg|\.jpeg)/;
     const brandLinks = [...brandBlock.querySelectorAll('a')];
-    const image = brandLinks.find((brandLink) => imgRegex.test(brandLink.href));
-    const link = brandLinks.find((brandLink) => !imgRegex.test(brandLink.href));
+    const image = brandLinks.find((brandLink) => imgRegex.test(brandLink.href)
+      || imgRegex.test(brandLink.textContent));
```

Comment:
Wdyt about checking if the image is an SVG? And then conditionally calling either `getSVG` or `getIMG` instead of putting both in one method?

What about `feds-navLink-image` and the logo - do they need that logic as well or is the brand special?

Reply:
The brand block is definitely a special use-case. Not only is it already used by all consumers, but each have authored it in a slightly different way. We promised backwards compatibility, so this aims to be as inclusive as possible without being too particular. The link images are usually added directly as PNG/JPG. The logo needs to be refactored (https://jira.corp.adobe.com/browse/MWPW-131183) and it will be based on logic from this brand feature.

I actually did try to split up the code, but due to how the authoring is done today by different consumers, that approach actually ended up having more checks scattered over multiple functions, so I decided to keep it all in one place. Additionally, since the Logo feature needs to be refactored too, we might be able to leverage `getBrandImage` there too.